src/app/modules/OpenshiftStreams/components/StreamsTableConnected/StreamsTableConnected.tsx

```diff
@@ -362,10 +365,19 @@ export const StreamsTableConnected: FunctionComponent<StreamsTableProps> = ({
     setSearchParam,
   ]);
 
+  const mounted = useRef(false);
+
+  useEffect(() => {
+    if (mounted.current === false || shouldRefresh) {
+      mounted.current = true;
+      fetchKafkas();
+    }
+  }, [fetchKafkas, shouldRefresh]);
+
+  // refresh the data when interacting with the UI in a way that will make the displayed data change
   useEffect(() => {
-    setKafkaDataLoaded(false);
-    fetchKafkas();
-  }, [auth, page, perPage, filteredValue, orderBy, fetchKafkas]);
+    refreshKafkasAfterAction();
+  }, [page, perPage, orderBy, searchParams, refreshKafkasAfterAction]);
```

Comment:
we also need `filteredValue` dependency for searching otherwise searching  will not work.